src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/composer/Composer.kt

```diff
@@ -311,6 +317,11 @@ class Composer(
         if (!inlineCommentsCollector.isEmpty()) {
             node.inLineComments = inlineCommentsCollector.consume()
         }
+        if (node.hasMergeTag) {
+            val updatedValue = mergeUtils.flatten(node)
+            node.value = updatedValue
+            node.hasMergeTag = false
```

Comment:
suggestion: What if we hide that behind an instance method, like

`node.merged(updatedValue)`

In this case, we can make setters for `value` and `hasMergeTag` private